src/hdf5plugin/_filters.py

```diff
@@ -466,34 +465,39 @@ class SZ(_FilterRefClass):
     filter_name = "sz"
     filter_id = SZ_ID
 
-    def __init__(self, absolute=None, relative=None, pointwise_relative=1e-05):
+    def __init__(self, absolute=None, relative=None, pointwise_relative=None):
+        if (absolute, relative, pointwise_relative).count(None) < 2:
+            raise TypeError("hdf5plugin.SZ() takes at most one not None argument")
+
         # Get SZ encoding options
         if absolute is not None:
             sz_mode = 0
-            parameter = absolute
         elif relative is not None:
             sz_mode = 1
-            parameter = relative
-        elif pointwise_relative is not None:
-            sz_mode = 10
-            parameter = pointwise_relative
         else:
-            raise TypeError("One of the options need to be provided: absolute, relative or pointwise_relative.")
+            sz_mode = 10
+            if pointwise_relative is None:
+                pointwise_relative = 1e-5
```

Comment:
There looks to be a lot more modes:
https://github.com/silx-kit/hdf5plugin/blob/daf5703f744d4ba3d6ab4c9980c99cd4e5c96bcb/src/SZ/sz/include/defines.h#L29-L41

So I propose to change a bit the arguments handling to force only one argument for now to prepare for adding support of additional modes.
We can later add for example `ABS_AND_REL` when both `absolute` and `relative` are provided.
But I expect an extra `mode` argument will be needed for the `OR` modes (e.g., `ABS_OR_REL`).

Reply:
I would still leave a working default as in the previous code. All our filters are working without modification.

Reply:
I kept the same default as before:
```python
            if pointwise_relative is None:
                pointwise_relative = 1e-5
```

Reply:
OK. I misunderstood the assert.